cli/azd/internal/appdetect/pom.go

```diff
@@ -2,32 +2,40 @@ package appdetect
 
 import (
 	"bufio"
+	"context"
 	"encoding/xml"
 	"fmt"
+	"log"
+	"log/slog"
 	"os"
 	"os/exec"
 	"path/filepath"
-	"regexp"
 	"strings"
 )
 
 // pom represents the top-level structure of a Maven POM file.
 type pom struct {
-	XmlName              xml.Name             `xml:"project"`
-	Parent               parent               `xml:"parent"`
-	Modules              []string             `xml:"modules>module"` // Capture the modules
-	Properties           Properties           `xml:"properties"`
-	Dependencies         []dependency         `xml:"dependencies>dependency"`
-	DependencyManagement dependencyManagement `xml:"dependencyManagement"`
-	Build                build                `xml:"build"`
-	path                 string
+	XmlName                 xml.Name             `xml:"project"`
+	Parent                  parent               `xml:"parent"`
+	GroupId                 string               `xml:"groupId"`
+	ArtifactId              string               `xml:"artifactId"`
+	Version                 string               `xml:"version"`
+	Modules                 []string             `xml:"modules>module"` // Capture the modules
+	Properties              Properties           `xml:"properties"`
+	Dependencies            []dependency         `xml:"dependencies>dependency"`
+	DependencyManagement    dependencyManagement `xml:"dependencyManagement"`
+	Build                   build                `xml:"build"`
+	pomFilePath             string
+	propertyMap             map[string]string
+	dependencyManagementMap map[string]string
```

Comment:
1. Above 2 maps are helper to improve the efficiency. Because it's often to read / update. by key.
2. I didn't add add map for Dependencies because it has much lower possibility of read / update by key.